src/packages/core/src/server.ts

```diff
@@ -185,6 +185,17 @@ export class Server<
       host = null;
     }
     const callbackIsFunction = typeof callback === "function";
+
+    // Method signature specifies port: number, but we parse a non-number if provided
+    // Matches the behaviour of http.server.listen https://www.w3schools.com/nodejs/met_server_listen.asp
+    if (isNaN(port)) {
+      const err = new Error(`Provided port is not a valid value: ${port}.`);
+      return callbackIsFunction
+        ? process.nextTick(callback!, err)
+        : Promise.reject(err);
+    }
+    const portNumber = typeof(port) === "number" ? port : parseInt(port);
+
```

Comment:
You probably want to do an interactive rebase in order to drop this commit. See https://stackoverflow.com/a/40926981 for a way to do this.